tests/Moq.Tests/Regressions/FluentMockVisitorIssues.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Xunit;
+
+namespace Moq.Tests.Regressions
+{
+	public class FluentMockVisitorIssues
+	{
+		[Fact]
+		public void WhenQueryingWithTwoIndexes_ThenSetsThemDirectly1()
+		{
+			var foo = Mock.Of<IFoo>(x => x[0].Value == "hello" && x[1].Value == "goodbye");
+
+			Assert.Equal("hello", foo[0].Value); // Fails here as foo[0] is the same object as foo[1] and foo[1].Value == "goodbye", Fixed!
```

Comment:
The _"fails here"_ comment will no longer make sense once you've fixed the issue. Please remove it. (With all tests, the basic assumption is that they will pass, so there should be no comment saying _"this fails"_. ;-)